xcube/util/jsonschema.py

```diff
@@ -46,6 +46,7 @@ def __init__(self,
                  title: str = None,
                  description: str = None,
                  examples: str = None,
+                 any_of: Sequence['JsonSchema'] = None,
```

Comment:
I have excluded the `anyOf`, `oneOf`, `allOf` keywords (https://json-schema.org/understanding-json-schema/reference/combining.html) from xcube by intention. They add complexity into the `from_instance` and `to_instance` method implementations. As far as I can see, you ignore the new `any_of` parameter in those implementations, which is not ok.

Reply:
> As far as I can see, you ignore the new `any_of` parameter in those implementations, which is not ok.

Yes, that was an oversight. How should I proceed? Add support for `any_of` to `from_instance` and `to_instance`, or remove `any_of` entirely? And if the latter, what to do about date specifications?

1. Only accept `date-time` format?
2. Only accept `date` format?
3. Deactivate JSON Schema format validation and handle it elsewhere?
4. Implement alternative formats for the date specifiers using some other JSON Schema mechanism that doesn't require `any_of`? But I don't know of any such mechanism (that's why I used `any_of`).

Reply:
Do you know if `date-time` includes `date`? I.e. is "2020-09-23" a valid datetime. In Pandas it is not. In xarray sel() "2020-09-23" converts into a datetime range "2020-09-23T00:00:00.000" to "2020-09-23T23:59:59.999".

Reply:
Ok, then we'll leave it as-is.

Reply:
I've added TODOs for anyOf handling in `from_instance` and `to_instance` and otherwise left it as is.